src/components/MaterialReservationCard.js

```diff
@@ -0,0 +1,169 @@
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+
+import Event from '@material-ui/icons/Event';
+import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
+import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
+
+import MaterialRulesDialog from '../components/MaterialRulesDialog';
+
+class MaterialReservationCard extends React.Component {
+
+  state = {
+    date: new Date(),
+    description: '',
+    items: [],
+    dateChosen: false,
+    dialogOpen: false,
+  }
+
+  handleChange = prop => event => {
+    this.setState({[prop]: event.target.value});
+  }
+
+  handleDateChange = value => {
+    this.setState({date: value});
+  }
+
+  handleBackButton = () => {
+    this.setState({dateChosen: false, items: []});
+  }
+
+  handleNextButton = () => {
+    this.setState({dateChosen: true});
+  }
+
+  handleDialogAccept = () => {
+    console.log('accept');
+    this.setState({dialogOpen: false});
+  }
+
+  handleDialogChange = dialogOpen => () => {
+    this.setState({dialogOpen: dialogOpen});
+  }
+
+  render() {
+    const { reservations, names } = this.props;
+
+    let maxDate = new Date();
+    maxDate.setMonth(maxDate.getMonth() + 4);
+
+    const dates = reservations.map(x => x.date);
+    const shouldDisableDate = (date) => {
+        return dates.some(taken => {
+            return date.getDate()  === taken.getDate() &&
+                   date.getMonth() === taken.getMonth() &&
+                   date.getYear()  === taken.getYear();
+        });
+    }
+
+    return (
+        <React.Fragment>
+          <Typography variant="body2" gutterBottom>
+            Nieuwe reservatie
+          </Typography>
+          <Grid container spacing={16}>
+            <Grid item xs={12}>
+              <DatePicker
+                disabled={this.state.dateChosen}
+                label="Datum"
+                fullWidth
+                InputLabelProps={{
+                  shrink: true,
+                }}
+                onChange={this.handleDateChange}
+                value={this.state.date}
+                disablePast
+                maxDate={maxDate}
+                shouldDisableDate={shouldDisableDate}
+                format='dd/MM/yyyy'
+                InputProps={{
+                  endAdornment: <InputAdornment position="end">
+                                  <IconButton><Event /></IconButton>
+                                </InputAdornment>
+                }}
+                leftArrowIcon={<KeyboardArrowLeft />}
+                rightArrowIcon={<KeyboardArrowRight />}
+              />
+            </Grid>
+            { this.state.dateChosen ? 
+                <Grid item xs={12}>
+                  <FormControl fullWidth disabled={!this.state.dateChosen}>
+                    <InputLabel shrink htmlFor="select-multiple">Materiaal</InputLabel>
```

Comment:
Hier was ik even verward, het is niet echt duidelijk dat je meerdere dingen tegelijk aan het selecteren bent, en ik dacht dat het kapot was (omdat het niet sloot) Eventueel kan je nog een vinkje ernaast proberen implementeren (want gewoon kleuren is niet genoeg), of een `+ nog iets` knop wanneer je iets gekozen hebt, of je kan gewoon maar 1 ding per reservatie selecteren. Als mensen 2 BBQ's willen huren moeten ze maar 2 reservaties plaatsen.

Reply:
resolved met checkboxes, uit vorige reservaties blijkt wel dat mensen meerdere items reserveren in 1